### ISSUE TITLE
SSSD should perform input-validation for values in sssd.conf

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#420]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/416
- Created at 2010-03-10 17:48:19 by [sgallagh](https://pagure.io/user/sgallagh)
- Closed as Duplicate
- Assigned to [dpal](https://pagure.io/user/dpal)
---

The SSSDConfig API should be extended to verify values before they are saved. Right now, we are limited to verifying that the type matches, but this is not very efficient.

The schema for the sssd.api.conf should be expanded to provide a perl-style regular expression that must be matched for the option to be saved. This would allow us to handle complicated conditions, as well as enumerated types.

I'm proposing this for SSSD 1.2.

### Comments

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2010-03-16 14:09:17](https://pagure.io/SSSD/sssd/issue/416#226209)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.2

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2010-03-16 14:40:00](https://pagure.io/SSSD/sssd/issue/416#226210)**

Rather than just regular expressions, it should probably be more generic. Range validation for integral types, enumerations, etc.

milestone: SSSD 1.2 => SSSD 1.3
summary: SSSDConfig API should perform regular-expression matching for values => SSSDConfig API should perform input-validation for values

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2010-03-31 16:23:45](https://pagure.io/SSSD/sssd/issue/416#226211)**

I'm extending this to encompass a series of bugs against the SSSD itself as well. Right now, we're too accepting of unknown and poorly-formed entries in the sssd.conf in both the SSSDConfig API and the daemon itself.

My proposed approach will be to redesign the sssd.api.conf as an XML document that will contain validation information for the options it contains. The daemon and the SSSDConfig API should both refer to this file (and its drop directory for provider options) in order to validate the config file.

Related Bugzilla Bugs:
https://bugzilla.redhat.com/show_bug.cgi?id=578511
https://bugzilla.redhat.com/show_bug.cgi?id=578503
https://bugzilla.redhat.com/show_bug.cgi?id=578498

component: SSSDConfig API => SSSD
owner: sgallagh => somebody
summary: SSSDConfig API should perform input-validation for values => SSSD should perform input-validation for values in sssd.conf
tests: 0 => 1

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2010-03-31 16:30:29](https://pagure.io/SSSD/sssd/issue/416#226212)**

Fields changed

owner: somebody => sgallagh

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2010-07-20 15:30:12](https://pagure.io/SSSD/sssd/issue/416#226213)**

Fields changed

milestone: SSSD 1.5.0 => SSSD 1.6.0

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-02-07 15:21:38](https://pagure.io/SSSD/sssd/issue/416#226214)**

Fields changed

coverity:  => 
milestone: SSSD 1.6.0 => SSSD 1.7.0
owner: sgallagh => dpal
upgrade:  => 0

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-07-22 22:21:39](https://pagure.io/SSSD/sssd/issue/416#226215)**

Fields changed

milestone: SSSD 1.8.0 => SSSD 1.9.0
patch:  => 0

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-16 16:46:55](https://pagure.io/SSSD/sssd/issue/416#226216)**

Fields changed

blockedby:  => 
blocking:  => 
milestone: SSSD 1.9.0 => SSSD Deferred
rhbz:  =>

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-19 03:18:22](https://pagure.io/SSSD/sssd/issue/416#226217)**

Fields changed

rhbz:  => 0

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-05-14 14:51:34](https://pagure.io/SSSD/sssd/issue/416#226218)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=578511 (*Fedora*)

rhbz: 0 => [https://bugzilla.redhat.com/show_bug.cgi?id=578511 578511]

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2013-11-04 07:17:30](https://pagure.io/SSSD/sssd/issue/416#226219)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=1025825 (*Red Hat Enterprise Linux 6*)

rhbz: [https://bugzilla.redhat.com/show_bug.cgi?id=578511 578511] => [https://bugzilla.redhat.com/show_bug.cgi?id=578511 578511], [https://bugzilla.redhat.com/show_bug.cgi?id=1025825 1025825]

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2014-03-20 15:48:14](https://pagure.io/SSSD/sssd/issue/416#226220)**

Linked to Bugzilla bug: https://bugzilla.redhat.com/show_bug.cgi?id=1077695 (*Red Hat Enterprise Linux 7*)

rhbz: [https://bugzilla.redhat.com/show_bug.cgi?id=578511 578511], [https://bugzilla.redhat.com/show_bug.cgi?id=1025825 1025825] => [https://bugzilla.redhat.com/show_bug.cgi?id=578511 578511], [https://bugzilla.redhat.com/show_bug.cgi?id=1025825 1025825], [https://bugzilla.redhat.com/show_bug.cgi?id=1077695 1077695]

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2015-04-24 15:27:39](https://pagure.io/SSSD/sssd/issue/416#226221)**

Closing as duplicate of #2269

changelog:  => 
design:  => 
design_review:  => 0
feature_milestone:  => 
fedora_test_page:  => 
mark:  => 0
resolution:  => duplicate
review:  => 0
selected:  => 
status: new => closed

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2017-02-24 14:33:10](https://pagure.io/SSSD/sssd/issue/416#226222)**

**Metadata Update from @sgallagh**:
- Issue assigned to dpal
- Issue marked as depending on: #1621
- Issue marked as depending on: #1622
- Issue set to the milestone: SSSD Patches welcome
